From c00313fbfa4956d246b1bbb2c2bd7b9dce56ccac Mon Sep 17 00:00:00 2001
From: Manny Chang <manny.chang@mediatek.com>
Date: Tue, 17 Aug 2021 18:23:29 +0800
Subject: [PATCH 1/1] [ALPS06064371] mtprintk: driver porting

support mtprink feature

Change-Id: I5e1b6372e23001c300bccae71e2d03b884feb6ef
Signed-off-by: Manny Chang <manny.chang@mediatek.com>
CR-Id: ALPS06064371
Feature: Mediatek Printk Control(mtprintk)
---
 kernel/Makefile                               |   1 +
 kernel/printk/printk.c                        |   7 +
 2 files changed, 8 insertions(+)

diff --git a/kernel/Makefile b/kernel/Makefile
index 54a4b3311786..e6a3815f9bf2 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -3,6 +3,7 @@
 # Makefile for the linux kernel.
 #
 
+subdir-ccflags-y += -I$(srctree)/drivers/misc/mediatek/include
 obj-y     = fork.o exec_domain.o panic.o \
 	    cpu.o exit.o softirq.o resource.o \
 	    sysctl.o sysctl_binary.o capability.o ptrace.o user.o \
diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index d74eabff50e6..bc551777c6ef 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -62,6 +62,8 @@
 #include "braille.h"
 #include "internal.h"
 
+#include "mt-plat/mtk_printk_ctrl.h"
+
 #ifdef CONFIG_PRINTK_PREFIX_ENHANCE
 static DEFINE_PER_CPU(char, printk_state);
 #endif
@@ -1844,6 +1846,11 @@ static void call_console_drivers(const char *ext_text, size_t ext_len,
 		return;
 
 	for_each_console(con) {
+#ifdef CONFIG_PRINTK_MTK_UART_CONSOLE
+		/* if uart printk disabled */
+		if (!mt_get_uartlog_status() && (con->flags & (CON_CONSDEV | CON_BOOT)))
+			continue;
+#endif
 		if (exclusive_console && con != exclusive_console)
 			continue;
 		if (!(con->flags & CON_ENABLED))
-- 
2.18.0

